client/dashboard/sites/site/environment-switcher.tsx

```diff
@@ -129,13 +142,41 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
 		enabled: !! otherEnvironmentSiteId,
 	} );
 
+	const stagingSiteId = site.is_wpcom_staging_site ? site.ID : otherEnvironmentSiteId;
+	const { data: isStagingSiteDeleting } = useQuery( {
+		...isDeletingStagingSiteQuery( stagingSiteId ?? 0 ),
+		enabled: !! stagingSiteId,
+	} );
+
+	const productionSiteId = site.is_wpcom_staging_site
+		? site.options?.wpcom_production_blog_id ?? 0
+		: site.ID;
```

Comment:
Here we could consider getting the `productionSiteId` with the help of `getProductionSiteId` (from `/utils/site-staging-site`).